### ISSUE TITLE
a question about inference

### ISSUE THREAD

**Issue by Birdylx** [PaddlePaddle/PaddleSeg#1080]
Hi, thanks for your work. I had a question about inference in this codebase. [line](https://github.com/PaddlePaddle/PaddleSeg/blob/9042ce16c4c0e019e3c24096b42ce81f049efff4/paddleseg/core/infer.py#L225) shows that you take argmax first on logits, then upsample to img's original shape using 'nearest' mode. why not upsample logits first using 'bilinear' mode, then take argmax on upsampled logits. In my experiments, with my own dataset, the later inference method will always improve about 1 point mIoU, and moreover, the perceptual quality of segmentation improves significantly, and the boundary get more smooth, especially for small size img.

**Comment by LutaoChu** [comment]
Hi, due to historical reasons, we have ignored this phenomenon. At present, we have submitted PR for optimization. Thank you for your feedback！

**Comment by LutaoChu** [comment]
Our experiment shows that we can increase mIoU by 0.2 points, but you increase it by 1 point. What's the size of your image?

**Comment by Birdylx** [comment]
@LutaoChu Hi, I test it on my own dataset, I trained OCRNet with input size of 448, and inferenced with input size 448. Here are the results of two different inference methods

![0372_0001](https://user-images.githubusercontent.com/29754889/121690776-db9afd80-caf8-11eb-8526-0f3c0f99dea1.jpg)
![0372_0001](https://user-images.githubusercontent.com/29754889/121690805-e48bcf00-caf8-11eb-84b0-80463dfa5a4c.png)
![0372_0001](https://user-images.githubusercontent.com/29754889/121690837-eb1a4680-caf8-11eb-8bd8-3556a68601e7.png)

as you can see, the later one has more smooth boundary, this often happens when the original image size is larger than 448, if the size is smaller than 448, there is no big difference. My dataset image's size are large, I guess this is the reason that I got 1 point improvement. Maybe you can test it on some public dataset like Cityscapes or ADE20K, I'm interested about it, please tell me the results if you test it.
Btw, other repo like [mmsegmentation](https://github.com/open-mmlab/mmsegmentation/blob/fb24bf54b69ad8e83098e4b18d789f25d0762874/mmseg/models/segmentors/encoder_decoder.py#L271) take resize first inference strategy

**Comment by Birdylx** [comment]
@LutaoChu Hi, did you test it? does it have any improvments?

**Comment by LutaoChu** [comment]
Hi, I see. It's in line with expectations. The larger the size of the original image, the more obvious the effect improvement is. I think no more testing is required.
